subprojects/docs/src/docs/userguide/rule_source.adoc

```diff
@@ -15,6 +15,13 @@
 [[rule_source]]
 = Implementing model rules in a plugin
 
+[CAUTION]
+====
+Rule based configuration link:https://blog.gradle.org/state-and-future-of-the-gradle-software-model[will be deprecated].
+New plugins should not use this concept.
+Instead, refer to <<custom_plugins.adoc#custom_plugins, Writing Custom Plugins>> chapter.
```

Comment:
I would prefer:

> Instead, use the standard approach described in the Writing Custom Plugins chapter.

That follows on more naturally from the previous sentence, since you want to say what plugin authors should be doing instead.